quesma/schema/registry.go

```diff
@@ -5,21 +5,29 @@ package schema
 import (
 	"quesma/logger"
 	"quesma/quesma/config"
-	"strings"
+	"quesma/util"
+	"sync"
 )
 
 type (
 	Registry interface {
 		AllSchemas() map[TableName]Schema
 		FindSchema(name TableName) (Schema, bool)
 		UpdateDynamicConfiguration(name TableName, table Table)
+		UpdateFieldEncodings(encodings map[FieldEncodingKey]string)
+	}
+	FieldEncodingKey struct {
```

Comment:
as chatted on Slack, I think what really intricate with this structure at this moment is that I was trying to come up with a better naming - because we should be super precise which is public and which is internal fieldname. At this point we should ditch the generic "fieldName" I think and be super specific about what kind of field name it is specifically - public-facing or internal.

Then I realized it may be better for this encodings to actually live in the `Schema` object where we already "kind-of" keep this information? `schema.go`:
![image (16)](https://github.com/user-attachments/assets/81a85ae3-81e4-4afc-a349-f78c1b793022)


If that brings too much hassle within this already huge PR, I'm good with keeping this duplication, but I'd advice putting a big fat warning saying "TODO HEY THIS IS SOMETHING WE REALLY NEED TO CLEANUP" 😉

Reply:
I changed signature `UpdateFieldEncodings(encodings map[FieldEncodingKey]EncodedFieldName)` and refined type and also add a comment for further improvements